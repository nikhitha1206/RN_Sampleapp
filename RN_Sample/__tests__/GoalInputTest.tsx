import { render, fireEvent, cleanup } from "@testing-library/react-native";
import React from 'react';
import { Button } from "react-native";
import renderer from 'react-test-renderer';
import GoalInput from '../components/GoalInput'
import { configure ,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Modal } from "react-native";

configure({ adapter: new Adapter() });

//Failing to call cleanup when you've called render could result in a memory leak and tests which are not "idempotent" (which can lead to difficult to debug errors in your tests).
afterEach(cleanup);

describe("render",()=>{
    const onEventMock = jest.fn();
    test('renders correctly', () => {
        const tree = renderer.create(<GoalInput />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    test("test presence of input field",()=>{
        const { getByPlaceholderText } = render(<GoalInput />);
        const input = getByPlaceholderText("Course Goal");
        expect(input).not.toBeNull();
    })  
    test("test setting value to input field",()=>{
        // const { getByPlaceholderText } = render(<GoalInput handleChangeText={onEventMock} />);
        // const input = getByPlaceholderText("Course Goal");
        // fireEvent.changeText(input, "abc");

     
      //expect(onEventMock.mockName).toHaveBeenCalledTimes(1);
    })
    test("wheater it is modal component",()=>{
        const wrapper=shallow(<GoalInput />)
        expect(wrapper.type()).toEqual(Modal)
    })
    test("wheater add button present or not",()=>{
        const { getByText} = render(<GoalInput />);
        const btn = getByText("ADD");
        expect(btn).not.toBeNull();
    })
    test("wheater cancel button present or not",()=>{
        const { getByText} = render(<GoalInput />);
        const btn = getByText("CANCEL");
        expect(btn).not.toBeNull();
    })
    test("number of buttons test",()=>{
        const wrapper=shallow(<GoalInput />)
        console.log(wrapper)
        expect(wrapper.find("Button")).toHaveLength(2)
    })
})