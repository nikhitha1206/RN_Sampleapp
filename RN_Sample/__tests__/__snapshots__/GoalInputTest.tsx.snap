// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Course Goal"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "margin": 10,
          "padding": 10,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "60%",
        }
      }
    >
      <View
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "red",
                  },
                ]
              }
            >
              CANCEL
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              ADD
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
